modules/islandora_advanced_search/src/Plugin/Block/SearchResultsPagerBlock.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+namespace Drupal\islandora_advanced_search\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Render\Markup;
+use Drupal\Core\Url;
+use Drupal\islandora_advanced_search\AdvancedSearchQuery;
+use Drupal\views\Entity\View;
+use Drupal\views\Plugin\views\pager\SqlBase;
+use Drupal\views\ViewExecutable;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Provides a 'AjaxViewBlock' block.
+ *
+ * @Block(
+ *  id = "islandora_advanced_search_result_pager",
+ *  deriver = "Drupal\islandora_advanced_search\Plugin\Block\SearchResultsPagerBlockDeriver",
+ *  admin_label = @Translation("Search Results Pager"),
+ *  category = @Translation("Islandora"),
+ * )
+ */
+class SearchResultsPagerBlock extends BlockBase implements ContainerFactoryPluginInterface {
+  use ViewAndDisplayIdentifiersTrait;
+
+  /**
+   * The clone of the current request object.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request
+   */
+  protected $request;
+
+  /**
+   * Construct a FacetBlock instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   A request object for the current request.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, Request $request) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->request = clone $request;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('request_stack')->getMasterRequest(),
```

Comment:
As per discussion in Slack this is 7.3 specific feature: https://wiki.php.net/rfc/trailing-comma-function-calls.

Reply:
Sorted